addons/point_of_sale/models/pos_order.py

```diff
@@ -25,6 +27,56 @@ class PosOrder(models.Model):
     _description = "Point of Sale Orders"
     _order = "date_order desc, name desc, id desc"
 
+    class PoSOrderData:
+        """
+        Convenience class which allow to easily compare 2 "orders datas".
+        This can be :
+         - an env["pos.order"] record (using `from_order_model_record`)
+         - an order raw data list coming from `create_from_ui` (using `from_order_data`)
+        """
+        def __init__(self, order_payment_value: list, order_payment_precision_rounding: int, order_lines_len: int, order_received_lines: list):
+            self.order_payment_value = order_payment_value
+            self.order_payment_precision_rounding = order_payment_precision_rounding
+            self.order_lines_len = order_lines_len
+            self.order_received_lines = order_received_lines
```

Comment:
We are in the context of 'order' here so I guess it's better to remove the `order_` prefix. E.g. just use `payment_value` instead of `order_payment_value`.